Tasks/Gradle/gradletask.ts

```diff
@@ -145,17 +144,23 @@ async function execBuild() {
     gb.exec()
         .then(function (code) {
             gradleResult = code;
+            tl.debug(`exit code: ${code}`);
             publishTestResults(publishJUnitResults, testResultsFiles);
             publishCodeCoverage(isCodeCoverageOpted);
             return processCodeAnalysisResults();
         })
         .then(() => {
-            tl.exit(gradleResult);
+            tl.debug(`Gradle result: ${gradleResult}`);
+            if (gradleResult === 0) {
+                tl.setResult(tl.TaskResult.Succeeded, "Build succeeded.");
+            } else {
+                tl.setResult(tl.TaskResult.Failed, "Build failed.");
+            }
         })
         .fail(function (err) {
             console.error(err);
             tl.debug('taskRunner fail');
-            tl.exit(1);
+            tl.setResult(tl.TaskResult.Failed, err);
```

Comment:
If I left tl.exit() in there, I always get a `[TypeError: tl.exit is not a function]`.  

I don't see this error in Maven.  Need proper fix later.